src/main/java/org/opentripplanner/middleware/controllers/api/TripHistoryController.java

```diff
@@ -27,46 +34,96 @@
 
     /**
-     * return JSON representation of a user's trip request history based on provided parameters
+     * Return JSON representation of a user's trip request history based on provided parameters.
+     * Only the user id is required.
      */
-    public static String getTripRequests(Request request, Response response, TypedPersistence<TripRequest> tripRequest, String expectedDatePattern) {
-        response.type("application/json");
+    public static String getTripRequests(Request request, Response response, TypedPersistence<TripRequest> tripRequest) {
 
-        DateTimeFormatter expectDateFormat = new DateTimeFormatterBuilder().appendPattern(expectedDatePattern)
-            .parseDefaulting(ChronoField.NANO_OF_DAY, 0)
-            .toFormatter()
-            .withZone(ZoneId.systemDefault());
+        final String userId = HttpUtils.getParamFromRequest(request, USER_ID_PARAM_NAME, false);
 
-        String userId = HttpUtils.getParamFromRequest(request, USER_ID_PARAM_NAME);
         int limit = DEFAULT_LIMIT;
 
-        String param = null;
+        String paramLimit = null;
         try {
-            param = HttpUtils.getParamFromRequest(request, LIMIT_PARAM_NAME);
-            limit = Integer.parseInt(param);
+            paramLimit = HttpUtils.getParamFromRequest(request, LIMIT_PARAM_NAME, true);
+            if (paramLimit != null) {
+                limit = Integer.parseInt(paramLimit);
+                if (limit <= 0) {
+                    limit = DEFAULT_LIMIT;
+                }
+            }
         } catch (NumberFormatException e) {
-            LOG.error("Unable to parse {} : {}. Using default limit: {}", LIMIT_PARAM_NAME, param, DEFAULT_LIMIT, e);
+            LOG.error("Unable to parse {} value of {}. Using default limit: {}", LIMIT_PARAM_NAME, paramLimit, DEFAULT_LIMIT, e);
         }
 
-        LocalDate fromDate = HttpUtils.getDateFromRequestParam(request, expectDateFormat, expectedDatePattern, FROM_DATE_PARAM_NAME);
-        LocalDate toDate = HttpUtils.getDateFromRequestParam(request, expectDateFormat, expectedDatePattern, TO_DATE_PARAM_NAME);
+        String paramFromDate = HttpUtils.getParamFromRequest(request, FROM_DATE_PARAM_NAME, true);
+        Date fromDate = getDate(request, FROM_DATE_PARAM_NAME, paramFromDate, LocalTime.MIN);
 
-        LocalDateTime fromStartOfDay = fromDate.atTime(LocalTime.MIN);
-        LocalDateTime toEndOfDay = toDate.atTime(LocalTime.MAX);
-
-        Bson filter = Filters.and(gte(TRIP_REQUEST_DATE_CREATED_FIELD_NAME, Date.from(fromStartOfDay.atZone(ZoneId.systemDefault()).toInstant())),
-            lte(TRIP_REQUEST_DATE_CREATED_FIELD_NAME, Date.from(toEndOfDay.atZone(ZoneId.systemDefault()).toInstant())),
-            eq(TRIP_REQUEST_USER_ID_FIELD_NAME, userId));
+        String paramToDate = HttpUtils.getParamFromRequest(request, TO_DATE_PARAM_NAME, true);
+        Date toDate = getDate(request, TO_DATE_PARAM_NAME, paramToDate, LocalTime.MAX);
```

Comment:
Suggestion: rename `LocalTime.MAX` to `LocalTime._23H59M`

Reply:
@binh-dam-ibigroup I don't understand your suggestion LocalTime._23H59M? This is not an option under java.time.LocalTime. I think LocalTime.MAX which provides 23:59:59 is ideal here.

Reply:
It is fine to keep it that way... I actually thought you created those variables!